app/src/main/java/pl/edu/zut/mad/appzut/fragments/ScheduleFragment.java

```diff
@@ -28,57 +34,110 @@
 import pl.edu.zut.mad.appzut.network.ScheduleEdzLoader;
 import pl.edu.zut.mad.appzut.utils.DateUtils;
 
-public class ScheduleFragment extends Fragment implements BaseDataLoader.DataLoadedListener<Schedule> {
+public class ScheduleFragment extends Fragment
+        implements BaseDataLoader.DataLoadedListener<Schedule>, CalendarFragment.CalendarListener,
+                   AppBarLayout.OnOffsetChangedListener {
 
-    private ScheduleEdzLoader scheduleLoader;
-    private Unbinder unbinder;
+    private static final String SELECTED_DATE = "selected_date";
+    private static final String EMPTY_TOOLBAR_TITLE = " ";
+    private static final float COLLAPSED_PERCENT_SHOW_TITLE = 0.65f;
+    private static final int FIRST_DAY_INDEX = 0;
+    private static final int LAST_DAY_INDEX = 6;
     @BindView(R.id.schedule_main) View scheduleWrapper;
-    @BindView(R.id.tabs) TabLayout tabLayout;
-    @BindView(R.id.pager) ViewPager pager;
+    @BindView(R.id.pager) ViewPager viewPager;
     @BindView(R.id.schedule_unavailable) View scheduleUnavailableWrapper;
     @BindView(R.id.import_from_edziekanat) TextView importFromEdziekanatButton;
     @BindView(R.id.loading_indicator) ProgressBar loadingIndicator;
+    private Unbinder unbinder;
+    private Toolbar toolbar;
+    private CollapsingToolbarLayout toolbarLayout;
+    private CalendarFragment calendarFragment;
+    private ScheduleEdzLoader scheduleLoader;
+    private SchedulePagerAdapter pagerAdapter;
+    private List<Date> currentWeekDates;
+    private String selectedDateString;
+    private boolean isBarTitleShow = false;
+    private int barShowHeight = -1;
+    private String[] weekDaysNames;
 
     @Override
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        readSavedSelectedDateString(savedInstanceState);
+        initWeekDaysNames();
+    }
+
+    private void readSavedSelectedDateString(Bundle savedInstanceState) {
+        if (savedInstanceState != null) {
+            selectedDateString = savedInstanceState.getString(SELECTED_DATE);
+        } else {
+            selectedDateString = DateUtils.convertDateToString(new Date());
+        }
+    }
+
+    private void initWeekDaysNames() {
+        weekDaysNames = getActivity().getResources().getStringArray(R.array.week_days);
```

Comment:
It's odd that you've left this given you're ButterKnife fan

Sometimes calendar restores to wrong date but I don't have reliable reproduction, however that can go in separate pull

Anyway, seems okay to merge